internal/app/gateway/command/run.go

```diff
@@ -128,6 +139,17 @@ func run(command *cobra.Command, args []string) {
 	log.Println("[mashling] App Description: ", gateway.Description())
 
 	// Startup the configured gateway instance.
+	pingData := PingData{gateway.Version(), gateway.AppVersion(), gateway.Description()}
```

Comment:
I would move all of this into the Gateway. That way the gateway can start and stop these related services once it has started successfully or stopped.